### PR TITLE
chore: Refactor cloud config init url as env var

### PR DESCRIPTION
**Pull request by MoeMahhouk**
This PR changes the cloud init URL from hardcoded to a way to pass it as env variable during the build process

### CODE REVIEW

File: recipes-nodes/config-parser/config-parser.bb [hunk 1]
```diff
@@ -10,8 +10,23 @@ SRC_URI = "file://config_parser.sh \
...
 
-# Default value for the config URL
-CLOUD_INIT_CONFIG_URL ?= "http://169.254.169.254/metadata/instance/compute/userData?api-version=2021-01-01&format=text"
+python () {
+    # Check if CLOUD_INIT_CONFIG_URL is set in the environment or in local.conf
+    cloud_init_config_url = d.getVar('CLOUD_INIT_CONFIG_URL')
+    
+    if cloud_init_config_url is None:
+        # If not set, check the original environment
+        origenv = d.getVar("BB_ORIGENV", False)
```
Comment (Ruteri) [inline recipes-nodes/config-parser/config-parser.bb:19]: Hm, what's the `BB_ORIGENV`?

Comment (MoeMahhouk) [reply recipes-nodes/config-parser/config-parser.bb:19]: when you want to use env variable of the calling host that runs bitbake command, it fetches them from the BB_ORIGENV.

Without this, the only possible way to pass them as env variable is through the bitbake ecosystem.  Such as, writing them in local.conf or in the recipes etc...
But in our case, we want to pass them when building as build arguments from docker

File: recipes-nodes/config-parser/config-parser.bb [hunk 2]
```diff
@@ -10,8 +10,23 @@ SRC_URI = "file://config_parser.sh \
 
 S = "${WORKDIR}"
 
-# Default value for the config URL
-CLOUD_INIT_CONFIG_URL ?= "http://169.254.169.254/metadata/instance/compute/userData?api-version=2021-01-01&format=text"
+python () {
```
Comment (Ruteri) [inline recipes-nodes/config-parser/config-parser.bb:13]: Is this python script any different than the bb thing on the right?

Comment (MoeMahhouk) [reply recipes-nodes/config-parser/config-parser.bb:13]: Yes, it is fetching it now from the bitbake caller env variable if passed prior to calling bitbake.
This way, we can pass this as build argument in the docker build as well similar to what we are doing with the searcher ssh key in the bob project

**Review by Ruteri — APPROVED** [review]
We might want to consider a common python function that checks the env and the BB_ORIGENV to use everywhere we need to pass stuff from the enviroment. Otherwise this could be easy to miss in places, also the function is really verbose in callsite - could instead just be `envOr(FLAG, "default")`

**Pull request merged**